pkg/provisioner/ironic/validatemanagementaccess_test.go

```diff
@@ -95,3 +96,43 @@ func TestValidateManagementAccessCreateNode(t *testing.T) {
 	assert.NotEqual(t, "", host.Status.Provisioning.ID)
 	assert.Equal(t, ironic.CreatedNodes[0].UUID, host.Status.Provisioning.ID)
 }
+
+func TestValidateManagementAccessLinkExistingIronicNodeByMAC(t *testing.T) {
```

Comment:
It would be good to have another test for the failure path by explicitly creating a port with the wrong UUID as well.